packages/amplify_api/ios/Classes/Auth/FlutterAuthProvider.swift

```diff
@@ -78,11 +78,23 @@ class FlutterAuthProviders: APIAuthProviderFactory {
         return token ?? unknownError
     }
 
+    private let authProviders: Set<AWSAuthorizationType>
+
+    init(_ authProviders: [AWSAuthorizationType]) {
+        self.authProviders = Set(authProviders)
+    }
+
     override func oidcAuthProvider() -> AmplifyOIDCAuthProvider? {
+        guard authProviders.contains(.openIDConnect) else {
+            return nil
+        }
         return FlutterAuthProvider(type: .openIDConnect)
```

Comment:
iOS gets tripped up if these return anything but nil, they believe we've registered an auth provider and this triggers different flows within DataStore.